tests/test_tasks.py

```diff
@@ -64,6 +64,28 @@ def test_load_nox_module_not_found():
     assert tasks.load_nox_module(config) == 2
 
 
+def test_load_nox_module_works_when_needs_nox_requires_lower_version():
+    config = _options.options.namespace(
+        noxfile=os.path.join(RESOURCES, "noxfile_past.py")
+    )
+    noxfile_module = tasks.load_nox_module(config)
+    assert noxfile_module.needs_nox == "2001.1.1"
+
+
+def test_load_nox_module_returns_2_when_needed_version_is_higher():
+    config = _options.options.namespace(
+        noxfile=os.path.join(RESOURCES, "noxfile_future.py")
+    )
+    assert tasks.load_nox_module(config) == 2
+
+
+def test_load_nox_module_returns_2_when_noxfile_does_not_even_import():
```

Comment:
It isn't clear to me what this validates. Is this validating the ordering of errors? That we will mention that we have an invalid arg to the decorator before the version?